src/main/navigation/navigation.c

```diff
@@ -1152,9 +1157,29 @@ static navigationFSMEvent_t navOnEnteringState_NAV_STATE_RTH_INITIALIZE(navigati
     }
 }
 
+static void overrideRTHAtitudePreset(void)
+{
+    if (!navConfig()->general.flags.rth_alt_control_override) {
+        return;
+    }
+
+    if (!IS_RC_MODE_ACTIVE(BOXNAVALTHOLD)) {
```

Comment:
This not-logic is counter-intuitive. Please reverse.

Reply:
Ah OK, yes that's obvious now you mention it. I'll also move the function to after `void updateHomePosition(void) ` (around line 2500 ) so it doesn't break up the sequence of those RTH phases if that's OK with you.